client/me/purchases/payment-methods/existing-credit-card.tsx

```diff
@@ -0,0 +1,397 @@
+
+	const queryKey = [ 'tax-info-is-set', storedDetailsId ];
+
+	const { data } = useQuery<
+		{ tax_postal_code: string; tax_country_code: string; is_tax_info_set: boolean },
+		Error
+	>( queryKey, () => fetchTaxInfo( storedDetailsId ), {} );
+
+	const mutation = useMutation(
+		( mutationInputValues: { tax_postal_code: string; tax_country_code: string } ) =>
+			setTaxInfo(
+				storedDetailsId,
+				mutationInputValues.tax_postal_code,
+				mutationInputValues.tax_country_code
+			),
+		{
+			onMutate: ( onMutateInputValues: { tax_postal_code: string; tax_country_code: string } ) => {
+				// Optimistically update the toggle
+				queryClient.setQueryData( queryKey, {
+					tax_postal_code: onMutateInputValues.tax_postal_code,
+					tax_country_code: onMutateInputValues.tax_country_code,
+					is_tax_info_set: true,
+				} );
+				closeDialog();
+			},
+			onSuccess: ( onSuccessInputValues: {
+				tax_postal_code: string;
+				tax_country_code: string;
+			} ) => {
+				queryClient.setQueryData( queryKey, {
+					tax_postal_code: onSuccessInputValues.tax_postal_code,
+					tax_country_code: onSuccessInputValues.tax_country_code,
+					is_tax_info_set: true,
+				} );
+			},
+		}
+	);
+
+	const [ inputValues, setInputValues ] = useState( {
+		tax_postal_code: data?.tax_postal_code ?? '',
+		tax_country_code: data?.tax_country_code ?? '',
+	} );
+
+	const postalCodeValue = inputValues.tax_postal_code;
+	const countryCodeValue = inputValues.tax_country_code;
+
+	const updateTaxInfo = useCallback( () => {
+		mutation.mutate( inputValues );
+	}, [ mutation, inputValues ] );
+
+	const onChangeCountryCode = ( e: { target: { value: string } } ) => {
+		setInputValues( { ...inputValues, tax_country_code: e.target.value } );
+	};
+
+	const onChangePostalCode = ( e: { target: { value: string } } ) => {
+		setInputValues( { ...inputValues, tax_postal_code: e.target.value } );
+	};
+
+	const formRender = (
+		<form>
+			<div className="contact-fields payment-methods__tax-fields">
+				<RenderEditFormFields
+					postalCodeValue={ postalCodeValue }
+					countryCodeValue={ countryCodeValue }
+					onChangePostalCode={ onChangePostalCode }
+					onChangeCountryCode={ onChangeCountryCode }
+				/>
+			</div>
+		</form>
+	);
+
+	/* translators: %s is the last 4 digits of the credit card number */
+	const maskedCardDetails = sprintf( _x( '**** %s', 'Masked credit card number' ), last4 );
+
+	const taxInfoDisplay = joinNonEmptyValues( ', ', postalCodeValue, countryCodeValue );
+
+	return (
+		<Fragment>
+			<div>
+				<CardHolderName>{ cardholderName }</CardHolderName>
+				<CardDetails>{ maskedCardDetails }</CardDetails>
+				<span>{ `${ __( 'Expiry:' ) } ${ formatDate( cardExpiry ) }` }</span>
+				{ taxInfoDisplay ? (
+					<span className="existing-credit-card__tax-info-display">
+						<span className="existing-credit-card__tax-info-postal-country">
+							{ taxInfoDisplay }
+						</span>
+						<PaymentMethodEditButton
```

Comment:
I think that the UI of this `Edit` button may not be ideal. It's very subtle: the text is smaller than the surrounding text, it is lighter in color, and there's no underline or border which to me makes it feel a little out-of-place.

<img width="683" alt="Screen Shot 2022-03-09 at 4 37 13 PM" src="https://user-images.githubusercontent.com/2036909/157540692-302f6dc2-1b02-41f0-921c-2ece188cc7a4.png">

What if we added an underline as an affordance to make it clear that it's clickable like we do with the `Remove from cart` button in checkout? 

https://github.com/Automattic/wp-calypso/blob/5152f5f056421d148081a8de3514edd412d833a1/packages/wpcom-checkout/src/checkout-line-items.tsx#L224-L238

Reply:
The font size/color is the same as the Remove From Cart buttons already, but I've added CSS to add an underline.